packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.scss

```diff
@@ -6,220 +6,242 @@
-    color: $bw-color;
-  }
-}
+$ms-checkbox-highContrast-iconBoxSize: 16px !default;
 
+// Default state
 .root {
-  box-sizing: border-box;
-  color: $ms-color-neutralPrimary;
-  font-size: $ms-font-size-m;
-  font-weight: $ms-font-weight-regular;
-  min-height: 36px;
-  position: relative;
-}
-
-.textLabel {
-  font-size: $ms-font-size-m;
-  @include padding(0, 0, 0, 26px);
-  display: inline-block;
-  font-size: $ms-font-size-m;
-}
-
-// The hidden input
-.input {
-  position: absolute;
-  opacity: 0;
-  top: 8px
-}
-
-// The checkbox square container
-.label::before {
-  content: '';
-  display: inline-block;
-  border: 1px solid $ms-color-neutralTertiary;
-  width: $ms-checkbox-label-size;
-  height: $ms-checkbox-label-size;
-  font-weight: normal;
-  position: absolute;
-  box-sizing: border-box;
-  transition-property: background, border, border-color;
-  transition-duration: $ms-checkbox-transition-duration;
-  transition-timing-function: $ms-checkbox-transition-timing;
-}
-
-// The check mark icon
-.label::after {
-  @include ms-Icon--CheckMark;
-  font-family: 'FabricMDL2Icons';
-  display: none;
-  position: absolute;
-  font-weight: 900;
-  background-color: transparent;
-  font-size: $ms-font-size-s-plus;
-  top: 0;
-  color: $ms-color-white;
-  line-height: $ms-checkbox-label-size;
-  width: $ms-checkbox-label-size;
-  text-align: center;
-}
-
-// The checkbox label
-.label {
-  display: inline-block;
-  cursor: pointer;
-  margin-top: 8px;
-  position: relative;
-  vertical-align: top;
-  user-select: none;
-  min-width: $ms-checkbox-label-size;
-  min-height: $ms-checkbox-label-size;
-  line-height: $ms-checkbox-label-size;
-
-  &:hover {
-    &::before {
-      border-color: $ms-color-neutralSecondaryAlt;
-    }
-
-    .textLabel {
-      color: $ms-color-black;
-    }
-  }
-
-  &:focus {
-    &::before {
-      border-color: $ms-color-neutralSecondaryAlt;
-    }
-
-    &.labelIsDisabled::before {
-      border-color: $ms-color-neutralTertiaryAlt;
-    }
-
-    &.labelIsChecked::before {
-      border-color: $ms-color-themeDarkAlt;
-    }
-  }
-
-  &:active {
-    &::before {
-      border-color: $ms-color-neutralSecondaryAlt;
-    }
-
-    .textLabel {
-      color: $ms-color-black;
-    }
-  }
-
-  //== State: A checkbox is checked
-  //
-  &.labelIsChecked {
-    &::before {
-      border: 10px solid $ms-color-themePrimary;
-      background-color: $ms-color-themePrimary;
-
-      @media screen and (-ms-high-contrast: active), (-ms-high-contrast: black-on-white) {
-        display: none;
-      }
-    }
-
-    &::after {
-      display: block;
-
-      @media screen and (-ms-high-contrast: active), (-ms-high-contrast: black-on-white) {
-        height: $ms-checkbox-highContrast-iconBoxSize;
-        width: $ms-checkbox-highContrast-iconBoxSize;
-        line-height: $ms-checkbox-highContrast-iconBoxSize;
-      }
-
-      @include highContrast-border($ms-color-white, $ms-color-black);
+    overflow: hidden;
```

Comment:
Yes, unnest as much as possible because it reduces CSS specificity so customizations are easier to do. Unfortunately, that wins out over readability/organization.